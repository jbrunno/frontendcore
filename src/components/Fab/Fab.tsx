import React from 'react';
import { Fab as MuiFab, FabProps as MuiFabProps } from '@mui/material';

export type FabProps = MuiFabProps;

export function Fab({ children, ...props }: FabProps) {
  return <MuiFab {...props}>{children}</MuiFab>;
}

export default Fab;
