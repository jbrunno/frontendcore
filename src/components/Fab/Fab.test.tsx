import React from 'react';
import { render } from '@test-utils/component-testing';
import { Fab } from './Fab';

test('renders the Fab', () => {
  const { container } = render(<Fab />);
  expect(container).toMatchSnapshot();
});
