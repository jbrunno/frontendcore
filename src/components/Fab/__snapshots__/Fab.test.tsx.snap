// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Fab 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-height: 36px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 50%;
  padding: 0;
  min-width: 0;
  width: 56px;
  height: 56px;
  z-index: 1050;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
  color: rgba(0, 0, 0, 0.87);
  background-color: #e0e0e0;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:active {
  box-shadow: 0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12);
}

.emotion-0:hover {
  background-color: #d5d5d5;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: #e0e0e0;
  }
}

.emotion-0.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-0.Mui-disabled {
  color: #00000042;
  box-shadow: none;
  background-color: #0000001F;
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <button
    class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default emotion-0"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root emotion-1"
    />
  </button>
</div>
`;
